src/components/Publications.vue

```diff
@@ -336,7 +391,7 @@ export default {
 
     deptFilter: async function (documents) {
       if (this.department !== "" && this.department !== null) {
-        return this.documents.filter((document) => {
```

Comment:
I added a fix here because this filtering on the whole documents list instead of the filtered list.